### ISSUE TITLE
Weird "dump is not defined" error when running tests

### ISSUE THREAD

**Issue by karina-y** [RecordReplay/devtools#1178]
**Describe the bug**
this happened only once so i don't have too much detail. i ran the test breakpoints-04.js on my console-eval branch and received an error in my console.

here is the pr i noticed this on
https://github.com/RecordReplay/devtools/pull/1169#issuecomment-741122343

**Screenshots**
![image](https://user-images.githubusercontent.com/19538307/101551758-70644900-3966-11eb-9993-040f8b3fc7c2.png)

**Logs**
```
 ~/code/replay.code/devtools   console-eval  node test/run.js --pattern breakpoints-04.js --record-all              SIGINT(2) ↵  335  14:10:34
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html breakpoints-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html breakpoints-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html breakpoints-03.js
[0.005] Starting test test/harness.js http://localhost:8080/test/examples/doc_control_flow.html breakpoints-04.js
JavaScript warning: resource:///modules/BrowserGlue.jsm, line 4427: unreachable code after return statement
1607465440106	addons.webextension.doh-rollout@mozilla.org	WARN	Loading extension 'doh-rollout@mozilla.org': Reading manifest: Invalid extension permission: networkStatus
1607465440477	addons.xpi-utils	WARN	Add-on doh-rollout@mozilla.org is not compatible with application version.
1607465440477	addons.xpi-utils	WARN	Add-on doh-rollout@mozilla.org is not compatible with application version.
1607465440478	addons.xpi-utils	WARN	Add-on doh-rollout@mozilla.org is not compatible with application version.
1607465440494	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: mozillaAddons
1607465440494	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: telemetry
1607465440494	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: resource://pdf.js/
1607465440494	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: about:reader*
1607465440497	addons.xpi-utils	WARN	Add-on screenshots@mozilla.org is not compatible with application version.
1607465440497	addons.xpi-utils	WARN	Add-on screenshots@mozilla.org is not compatible with application version.
1607465440498	addons.xpi-utils	WARN	Add-on screenshots@mozilla.org is not compatible with application version.
1607465440513	addons.xpi-utils	WARN	Add-on webcompat@mozilla.org is not compatible with application version.
1607465440513	addons.xpi-utils	WARN	Add-on webcompat@mozilla.org is not compatible with application version.
1607465440513	addons.xpi-utils	WARN	Add-on webcompat@mozilla.org is not compatible with application version.
updateRecordingDriver Starting... [Driver /Users/karinay/Library/Application Support/Replay/macOS-recordreplay.so]
TestHarnessStart
updateRecordingDriver AlreadyOnLatestVersion
CloudReplayStatus
TestHarnessHasURLBar
TestHarnessLoadURL http://localhost:8080/test/examples/doc_control_flow.html
JavaScript error: resource:///modules/UrlbarEventBufferer.jsm, line 174: Error: Event keydown:13 already deferred!
JavaScript warning: https://replay.io/dist/main.js, line 453: Script terminated by timeout at:
e.exports@https://replay.io/dist/main.js:453:37267
N@https://replay.io/dist/main.js:429:217103
@https://replay.io/dist/main.js:429:217884
async*@https://replay.io/dist/main.js:429:217939
n@https://replay.io/dist/main.js:1:110
@https://replay.io/dist/main.js:1:906
@https://replay.io/dist/main.js:1:917

console.log: "LogRocket: script could not load. Check that you have a valid network connection."
ClickRecordingButton
ClickRecordingButton
TestHarnessExampleRecordingTabStarted
WaitingForMessage Example__Finished
WaitingForMessage Example__Finished
ButtonClicked
ButtonClicked
JavaScript error: , line 0: uncaught exception: undefined
Attach Pid 50689 BuildId macOS-gecko-20201204150619 Version linker-macOS-2923-dab613e3
[#0/1 0.000] SendMessage 1 Internal.createRecording
[#0/1 0.001] AttachBinary /Applications/Replay.app/Contents/MacOS/XUL
Mapped /Applications/Replay.app/Contents/MacOS/XUL: 0x105ded000 [157147136 bytes]
[#0/1 0.058] AttachBinary /Applications/Replay.app/Contents/MacOS/libmozglue.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libmozglue.dylib: 0x10f3ce000 [598016 bytes]
[#0/1 0.059] AttachBinary /Applications/Replay.app/Contents/MacOS/libnss3.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libnss3.dylib: 0x105b4c000 [2748416 bytes]
JavaScript error: , line 0: uncaught exception: undefined
ClickRecordingButton
ClickRecordingButton
TestHarnessExampleRecordingTabFinished
TestHarnessWaitForDevtools
ButtonClicked
ButtonClicked
WaitForFinishedRecording
[#0/0 0.499] OnResponse 1 Internal.createRecording
CreateRecording 6935145e-94a2-444f-ac28-f522e1d75a46 http://localhost:8080/test/examples/doc_control_flow.html
[#0/1 0.513] FinishRecording
[#0/1 0.514] Sending all recording data...
[#0/1 0.514] SendMessage 2 Internal.addRecordingData
[#0/1 0.515] SendMessage 3 Internal.addRecordingDescription
SET_RECORDING_INFO
WaitForSavedRecording
FinishedRecording 6935145e-94a2-444f-ac28-f522e1d75a46
SET_RECORDING_INFO
ClickRecordingButton
ClickRecordingButton
TestHarnessViewerRecordingTabStarted
WaitingForMessage TestFinished
WaitingForMessage TestFinished
ButtonClicked
ButtonClicked
Attach Pid 50691 BuildId macOS-gecko-20201204150619 Version linker-macOS-2923-dab613e3
[#0/1 0.001] SendMessage 1 Internal.createRecording
[#0/1 0.002] AttachBinary /Applications/Replay.app/Contents/MacOS/XUL
Mapped /Applications/Replay.app/Contents/MacOS/XUL: 0x1079ed000 [157147136 bytes]
[#0/1 0.062] AttachBinary /Applications/Replay.app/Contents/MacOS/libmozglue.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libmozglue.dylib: 0x1025b3000 [598016 bytes]
[#0/1 0.062] AttachBinary /Applications/Replay.app/Contents/MacOS/libnss3.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libnss3.dylib: 0x102314000 [2748416 bytes]
[#0/1 0.366] AttachBinariesAfterOpen /Applications/Replay.app/Contents/MacOS/libsoftokn3.dylib
[#0/1 0.366] AttachBinary /Applications/Replay.app/Contents/MacOS/libsoftokn3.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libsoftokn3.dylib: 0x1281f4000 [299008 bytes]
[#0/1 0.367] AttachBinariesAfterOpen /Applications/Replay.app/Contents/MacOS/libfreebl3.dylib
[#0/1 0.367] AttachBinary /Applications/Replay.app/Contents/MacOS/libfreebl3.dylib
Mapped /Applications/Replay.app/Contents/MacOS/libfreebl3.dylib: 0x128285000 [528384 bytes]
[#0/0 0.472] OnResponse 1 Internal.createRecording
CreateRecording 67e85733-0ae6-40e5-8dc4-e3ef39e14f87 http://localhost:8080/view?id=6935145e-94a2-444f-ac28-f522e1d75a46&test=breakpoints-04.js
AddTestRecordingId 67e85733-0ae6-40e5-8dc4-e3ef39e14f87 200
[#0/0 2.479] OnResponse 2 Internal.addRecordingData
[#0/0 2.480] OnResponse 3 Internal.addRecordingDescription
[#0/1 2.480] Finished sending recording data.
[#0/1 2.480] Recording finished, exiting.
Redirecting call to abort() to mozalloc_abort

[#0/1 5.549] SendMessage 2 Internal.addRecordingData
[#0/1 5.807] SendMessage 3 Internal.addRecordingData
[#0/1 5.810] SendMessage 4 Internal.addRecordingData
JavaScript warning: http://localhost:8080/dist/main.js, line 189745: unreachable code after return statement
JavaScript warning: http://localhost:8080/dist/main.js, line 211429: unreachable code after return statement
JavaScript warning: http://localhost:8080/dist/main.js, line 258615: unreachable code after return statement
JavaScript warning: http://localhost:8080/dist/main.js, line 271180: unreachable code after return statement
console.log: "Download the React DevTools for a better development experience: https://fb.me/react-devtools"
console.log: "Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead."
[#0/1 6.553] SendMessage 5 Internal.addRecordingData
[#0/1 6.685] SendMessage 6 Internal.addRecordingData
console.log: "Warning: componentWillUpdate has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.\n\n* Move data fetching code or side effects to componentDidUpdate.\n* Rename componentWillUpdate to UNSAFE_componentWillUpdate to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.\n\nPlease update the following components: Lottie"
[#0/1 6.888] SendMessage 7 Internal.addRecordingData
console.log: "Socket Open"
[#0/1 7.832] SendMessage 8 Internal.addRecordingData
[#0/1 8.979] SendMessage 9 Internal.addRecordingData
console.log: "Starting addBreakpoint" "doc_control_flow.html" 10
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting rewindToLine" 10
console.log: "Finished rewindToLine in 1ms"
[#0/1 10.098] SendMessage 10 Internal.addRecordingData
console.log: "action[FETCHED_FRAMES].frames[0].this is of type ValueFront"
console.log: "action[MESSAGES_ADD].messages[0].parameters[0] is of type ValueFront"
console.log: "action[ADD_SCOPES].frame.this is of type ValueFront"
JavaScript warning: http://localhost:8080/dist/main.js, line 211429: unreachable code after return statement
console.log: "action[NODE_EXPAND].data.node.contents is of type ValueFront"
console.log: "action[NODE_PROPERTIES_LOADED].data.node.contents is of type ValueFront"
console.log: "Starting addBreakpoint" "doc_control_flow.html" 12
console.log: "Finished addBreakpoint in 1ms"
console.log: "Starting resumeToLine" 12
console.log: "Finished resumeToLine in 149ms"
[#0/1 12.290] SendMessage 11 Internal.addRecordingData
console.log: "Starting addBreakpoint" "doc_control_flow.html" 18
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 18
console.log: "Finished resumeToLine in 185ms"
[#0/1 14.123] SendMessage 12 Internal.addRecordingData
[#0/0 14.206] OnResponse 2 Internal.addRecordingData
console.log: "Starting addBreakpoint" "doc_control_flow.html" 20
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 20
console.log: "Finished resumeToLine in 225ms"
console.log: "Starting addBreakpoint" "doc_control_flow.html" 32
console.log: "Finished addBreakpoint in 0ms"
[#0/1 16.103] SendMessage 13 Internal.addRecordingData
console.log: "Starting resumeToLine" 32
console.log: "Finished resumeToLine in 273ms"
console.log: "Starting addBreakpoint" "doc_control_flow.html" 27
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 27
console.log: "Finished resumeToLine in 314ms"
[#0/1 19.016] SendMessage 14 Internal.addRecordingData
console.log: "Starting resumeToLine" 32
console.log: "Finished resumeToLine in 0ms"
console.log: "Starting resumeToLine" 27
console.log: "Finished resumeToLine in 1ms"
[#0/1 21.489] SendMessage 15 Internal.addRecordingData
1607465467442	addons.xpi-utils	WARN	Add-on doh-rollout@mozilla.org is not compatible with application version.
1607465467446	addons.xpi-utils	WARN	Add-on screenshots@mozilla.org is not compatible with application version.
1607465467449	addons.xpi-utils	WARN	Add-on webcompat@mozilla.org is not compatible with application version.
1607465467472	addons.xpi-utils	WARN	Add-on doh-rollout@mozilla.org is not compatible with application version.
1607465467472	addons.xpi-utils	WARN	Add-on screenshots@mozilla.org is not compatible with application version.
1607465467472	addons.xpi-utils	WARN	Add-on webcompat@mozilla.org is not compatible with application version.
console.log: "Starting addBreakpoint" "doc_control_flow.html" 42
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 42
console.log: "Finished resumeToLine in 466ms"
[#0/1 24.284] SendMessage 16 Internal.addRecordingData
console.log: "action[ADD_ASYNC_FRAMES].asyncFrames[0].this is of type ValueFront"
console.log: "Starting addBreakpoint" "doc_control_flow.html" 44
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 44
console.log: "Finished resumeToLine in 422ms"
[#0/1 26.943] SendMessage 17 Internal.addRecordingData
console.log: "Starting addBreakpoint" "doc_control_flow.html" 50
console.log: "Finished addBreakpoint in 1ms"
console.log: "Starting resumeToLine" 50
[#0/0 27.568] OnResponse 3 Internal.addRecordingData
console.log: "Finished resumeToLine in 468ms"
[#0/1 29.555] SendMessage 18 Internal.addRecordingData
console.log: "Starting addBreakpoint" "doc_control_flow.html" 54
console.log: "Finished addBreakpoint in 0ms"
console.log: "Starting resumeToLine" 54
console.log: "Finished resumeToLine in 525ms"
console.log: "TestFinished"
ClickRecordingButton
ClickRecordingButton
ButtonClicked
ButtonClicked
WaitForFinishedRecording
[#0/1 32.056] FinishRecording
[#0/1 32.058] SendMessage 19 Internal.addRecordingData
[#0/1 32.063] Sending all recording data...
[#0/1 32.063] SendMessage 20 Internal.addRecordingData
[#0/1 32.065] SendMessage 21 Internal.addRecordingDescription
SET_RECORDING_INFO
WaitForSavedRecording
TestHarnessViewerRecordingTabFinished
TestPassed
TestHarnessFinished
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
[#0/0 38.588] OnResponse 4 Internal.addRecordingData
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
[#0/0 42.809] OnResponse 5 Internal.addRecordingData
[#0/0 42.810] OnResponse 6 Internal.addRecordingData
[#0/0 42.810] OnResponse 7 Internal.addRecordingData
[#0/0 42.810] OnResponse 8 Internal.addRecordingData
[#0/0 42.810] OnResponse 9 Internal.addRecordingData
[#0/0 42.810] OnResponse 10 Internal.addRecordingData
[#0/0 42.810] OnResponse 11 Internal.addRecordingData
[#0/0 42.810] OnResponse 12 Internal.addRecordingData
[#0/0 42.810] OnResponse 13 Internal.addRecordingData
[#0/0 42.810] OnResponse 14 Internal.addRecordingData
[#0/0 42.810] OnResponse 15 Internal.addRecordingData
[#0/0 42.810] OnResponse 16 Internal.addRecordingData
[#0/0 42.811] OnResponse 17 Internal.addRecordingData
[#0/0 42.811] OnResponse 18 Internal.addRecordingData
[#0/0 43.038] OnResponse 19 Internal.addRecordingData
[#0/0 43.038] OnResponse 21 Internal.addRecordingDescription
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
console.error: (new Error("SessionFile is closed", "resource:///modules/sessionstore/SessionFile.jsm", 433))
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
CloudReplayStatus cloudError.label
CloudReplayStatus cloudReconnecting.label
Exiting due to channel error.
Exiting due to channel error.
[104.769] Test failed: Exited with code 1
::error ::Failure breakpoints-04.js https://replay.io/view?id=67e85733-0ae6-40e5-8dc4-e3ef39e14f87&test=breakpoints-04.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_debugger_statements.html breakpoints-05.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_prod_bundle.html breakpoints-06.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html stepping-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html stepping-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html stepping-03.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_blackbox.html stepping-04.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_minified.html stepping-05.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_async.html stepping-06.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_error.html console_warp-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_logs.html console_warp-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html console_eval.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_console.html console_messages.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_exceptions_bundle.html console_stacks.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html logpoint-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_basic.html logpoint-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_events.html logpoint-03.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_exceptions.html logpoint-04.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_objects.html object_preview-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_objects.html object_preview-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_preview.html object_preview-03.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_prod_bundle.html object_preview-05.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_rr_worker.html worker-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_inspector_basic.html inspector-01.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_inspector_basic.html inspector-02.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_inspector_styles.html inspector-03.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_inspector_styles.html inspector-04.js
Skipping test test/harness.js http://localhost:8080/test/examples/doc_inspector_sourcemapped.html inspector-05.js
[104.774] Had 1 test failures.
Failed test: breakpoints-04.js Exited with code 1
```

**Comment by jasonLaster** [comment]
closing as we haven't seen this for several weeks
